File: backend/kotlin/backend-kotlin-spring/src/main/kotlin/dev/reference/app/dto/Dtos.kt
package dev.reference.app.dto

import jakarta.validation.constraints.NotBlank
import jakarta.validation.constraints.Size

data class UserProfileResponse(
    val username: String,
)

data class ItemsResponse(
    val items: List<ItemDto>,
    val source: String,
)

data class RegisterRequest(
    @field:NotBlank @field:Size(min = 3, max = 64) val username: String,
    @field:NotBlank @field:Size(min = 6, max = 128) val password: String,
)

data class HealthResponse(
    val status: String,
    val service: String,
)

data class AuthResponse(
    val token: String,
    val username: String,
    val redirectUrl: String,
)

data class LoginRequest(
    @field:NotBlank @field:Size(min = 3, max = 64) val username: String,
    @field:NotBlank @field:Size(min = 6, max = 128) val password: String,
)

data class ItemDto(
    val id: Long,
    val name: String,
    val description: String,
)

File: backend/csharp/backend-csharp-aspnet/src/BackendCSharpAspnet/Api/Dtos.cs
using System.Text.Json.Serialization;

namespace Dev.Multistack.App.Api;

// Response bodies are types rather than dictionaries so the JSON field order matches
// the other reference backends exactly.

public sealed class HealthResponse
{
    [JsonPropertyOrder(0)]
    public required string Status { get; init; }

    [JsonPropertyOrder(1)]
    public required string Service { get; init; }
}

public sealed class ItemDto
{
    [JsonPropertyOrder(0)]
    public required long Id { get; init; }

    [JsonPropertyOrder(1)]
    public required string Name { get; init; }

    [JsonPropertyOrder(2)]
    public required string Description { get; init; }
}

public sealed class ItemsResponse
{
    [JsonPropertyOrder(0)]
    public required List<ItemDto> Items { get; init; }

    [JsonPropertyOrder(1)]
    public required string Source { get; init; }
}

public sealed class AuthResponse
{
    [JsonPropertyOrder(0)]
    public required string Token { get; init; }

    [JsonPropertyOrder(1)]
    public required string Username { get; init; }

    [JsonPropertyOrder(2)]
    public required string RedirectUrl { get; init; }
}

public sealed class ProfileResponse
{
    [JsonPropertyOrder(0)]
    public required string Username { get; init; }
}

public sealed class ErrorResponse
{
    [JsonPropertyOrder(0)]
    public required string Message { get; init; }
}
